test/e2e/aws_eip_for_nlb_test.go

```diff
@@ -0,0 +1,492 @@
+)
+
+func CreateAWSEIPs() ([]string, error) {
+	kubeClient, err := createClient()
+	if err != nil {
+		return nil, fmt.Errorf("test skipped: %v", err)
+	}
+	decodedAccessKeyID, decodedSecretAccessKey, err := getIDAndKey(kubeClient)
+	if err != nil {
+		return nil, fmt.Errorf("test skipped: %v", err)
+	}
+	if infraConfig.Status.PlatformStatus == nil {
+		return nil, fmt.Errorf("test skipped on nil platform")
+	}
+	region, err := getRegion()
+	if err != nil {
+		return nil, fmt.Errorf("test skipped: %v", err)
+	}
+	clusterName, err := getClusterName()
+	if err != nil {
+		return nil, fmt.Errorf("test skipped: %v", err)
+	}
+
+	// Set up an AWS session
+	sess, err := createSession(region, decodedAccessKeyID, decodedSecretAccessKey)
+	if err != nil {
+		return nil, fmt.Errorf("test skipped: %v", err)
+	}
+
+	// Create an EC2 service client
+	svc := ec2.New(sess)
+
+	vpcID, err := getVPCID(clusterName, svc)
+	if err != nil {
+		return nil, fmt.Errorf("test skipped: %v", err)
+	}
+
+	publicSubnets, err := getPublicSubnets(vpcID, svc)
+	if err != nil {
+		return nil, fmt.Errorf("test skipped: %v", err)
+	}
+
+	// Set up random seed
+	rand.Seed(time.Now().UnixNano())
+
+	// Set tag key and value
+	tagKeyEIP := fmt.Sprintf("NE1398-%s", os.Getenv("USER"))
+	tagValueEIP := clusterName
+
+	// Allocate EIPs
+	var allocationIDs []string
+	for i := 0; i < len(publicSubnets); i++ {
+		// Allocate EIP
+		result, err := svc.AllocateAddress(&ec2.AllocateAddressInput{
+			Domain: aws.String("vpc"),
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Error allocating EIP: %v", err)
+		}
+		allocationID := aws.StringValue(result.AllocationId)
+		fmt.Println("EIP allocated successfully. Allocation ID:", allocationID)
+
+		// Generate random name
+		randomName := fmt.Sprintf("%s-EIP-%d", clusterName, rand.Int())
+
+		// Tag EIP
+		_, err = svc.CreateTags(&ec2.CreateTagsInput{
+			Resources: []*string{result.AllocationId},
+			Tags: []*ec2.Tag{
+				{
+					Key:   aws.String("Name"),
+					Value: aws.String(randomName),
+				},
+				{
+					Key:   aws.String(tagKeyEIP),
+					Value: aws.String(tagValueEIP),
+				},
+			},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Error tagging EIP Allocation ID: %v", allocationID)
+		}
+		allocationIDs = append(allocationIDs, allocationID)
+	}
+	return allocationIDs, nil
+}
+
+func getClusterName() (string, error) {
+	var clusterName string
+	if len(infraConfig.Status.InfrastructureName) != 0 {
+		clusterName = infraConfig.Status.InfrastructureName
+	} else {
+		return "", fmt.Errorf("Cluster Name not found")
+	}
+	return clusterName, nil
+}
+
+func getRegion() (string, error) {
+	var region string
+	if infraConfig.Status.PlatformStatus.AWS != nil && len(infraConfig.Status.PlatformStatus.AWS.Region) != 0 {
+		region = infraConfig.Status.PlatformStatus.AWS.Region
+	} else {
+		return "", fmt.Errorf("Region not found")
+	}
+	return region, nil
+}
+
+func getIDAndKey(kubeClient client.Client) (string, string, error) {
```

Comment:
Thoughts on making a new `aws_util.go` file and putting all of these generic AWS API functions in it? I can see us reusing them, so forcing them to be in a common place would be helpful.

Reply:
OK.